### PR TITLE
[DM-29112] Add desired endpoint and show entire list

### PR DESCRIPTION
**Pull request by cbanek**

### CODE REVIEW

File: src/cachemachine/types.py [hunk 1]
```diff
@@ -102,6 +102,22 @@ def dump(self) -> List[Dict[str, str]]:
         return [asdict(i) for i in self]
 
 
+@dataclass
```
Comment (athornton) [inline src/cachemachine/types.py:105]: I'm confused; is the second list all images or all images XORed with desired images?

File: src/cachemachine/types.py [hunk 2]
```diff
@@ -102,6 +102,22 @@ def dump(self) -> List[Dict[str, str]]:
...
+    Contains two lists, one the images to pull, and the other
+    containing all the other images that could be pulled, but
+    won't be.
+    """
+
+    """Images that should be pulled."""
+    desired_images: DockerImageList
+
+    """Images that should end up in a sublist, but not be pulled."""
+    all_images: DockerImageList
```
Comment (athornton) [inline src/cachemachine/types.py:118]: If it really is XORed then maybe call it undesired_images?  If it is all images, then the comment should be something like "All images, whether or not they should be pulled."

**Review by athornton — CHANGES_REQUESTED** [review]
Generally looks good, but either the name or the comment for all_images is misleading.

**Comment by cbanek** [comment]
I didn't put the ones that will be in the first list in the all other list.  What name would you recommend?  Undesired doesn't seem like a great name.

**Comment by athornton** [comment]
> I didn't put the ones that will be in the first list in the all other list. What name would you recommend? Undesired doesn't seem like a great name.

I mean it's better than "repellent."  all_other_images would be better than all_images.  "not_to_be_pulled_images" seems a little wordy.

**Comment by cbanek** [comment]
Just made it contain all the images.

**Review by athornton — APPROVED** [review]
LGTM

**Pull request merged**